bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/builder/ThingBuilder.java

```diff
@@ -69,18 +70,22 @@ public ThingBuilder withLabel(@Nullable String label) {
 
     public ThingBuilder withChannel(Channel channel) {
         final Collection<Channel> mutableThingChannels = this.thing.getChannelsMutable();
+        validateChannelUIDs(Collections.singletonList(channel));
         ThingHelper.ensureUniqueChannels(mutableThingChannels, channel);
         mutableThingChannels.add(channel);
         return this;
     }
 
     public ThingBuilder withChannels(Channel... channels) {
-        ThingHelper.ensureUniqueChannels(channels);
-        this.thing.setChannels(new ArrayList<>(Arrays.asList(channels)));
+        List<Channel> newChannels = Arrays.asList(channels);
```

Comment:
Wouldn't it be better to call `return withChannels(Arrays.asList(channels));` instead of adding the same implementation twice? Or do I miss something?